src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationHandler.cs

```diff
@@ -183,7 +197,7 @@ private CookieOptions BuildCookieOptions()
                 return;
             }
 
-            var ticket = await AuthenticateOnceAsync();
+            var ticket = await HandleAuthenticateOnceAsync();
             try
             {
                 await ApplyCookie(ticket);
```

Comment:
This is the only place that ApplyCookie is called. Consider inlining it or refactoring it so that it can also be called from HandleSignIn